lib/utils/tls.go

```diff
@@ -54,6 +63,67 @@ type CertificateName struct {
 	OrganizationalUnit []string `json:"org_unit"`
 }
 
+// GenerateSelfSignedCert generates a self signed certificate that
+// is valid for given domain names and ips, returns PEM-encoded bytes with key and cert
+// Generates a certificate that is compatible with the MacOS requirements described at:
+// https://support.apple.com/en-us/HT210176
+func GenerateSelfSignedCert(hostNames []string) (*teleutils.TLSCredentials, error) {
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	notBefore := time.Now()
+	notAfter := notBefore.Add(time.Hour * 24 * 825) // 825 days or fewer is the required validity period for MacOS
```

Comment:
So I'm thinking, the new validity period for the certificate is just over 2 years and I think a lot of folks just go with never updating the certificate to their own, which means that it will stop working for them after 2 years. It will also affect not just the web UI but operations since various gravity commands also call HTTP APIs.

I wonder if gravity should also detect if the certificate is about to expire and reissue it. Or do you think we will migrate to something like K8s CSR API by then so it'll no longer be an issue? @knisbet @a-palchikov what do you guys think?

Reply:
I think it's a valid point and should not be too hard to implement even if we move to CSR APIs in the future.

Reply:
1. Yea, I think we'll need some way to rotate the certificate automatically. This will be tricky, because we probably don't want to replace a user provided certificate even if it's expired. 
    1. It really depends on how the client reacts to an expired self signed cert. If it produces a similar alert to a self signed cert, this only really needs to be manual/cli driven flow. If it's a hard error like happens to chrome currently, then it should be automatic sometime before expiry. Because requirements / browsers may change in the future, it's probably safer to automate though. In which case I'd probably even go even tighter than 825 days.
    1. For user provided certs, probably should have an alert that the cert is about to expire as well. Can treat this as a separate feature though.
    
2. I'm not as sure gravity operations will be effected on present builds. AFAIK we don't properly configure the trust store anywhere for the internal comms, and we have lots of hard code for setting insecure on all TLS connections which I haven't had the opportunity to fix. This is awful, but however that gets solved should account for the ability for certs to rotate, or use separate long term trust certs for m2m connections.
    1. My knee jerk thought is long term the internal comms should probably hang off their own internal CA/Cert that the web server presents using SNI, and you know, setup the trust anchors.